### ISSUE TITLE
Das submodule-update funktioniert nicht, wenn das gluon-Verzeichnis nicht existiert

### ISSUE THREAD

**Issue by Tronde** [FreifunkLippe/site-ffl#26]
Hallo @collimas,

ich habe deine E-Mail gelesen und eröffne diesen Issue, da wir hier besser zu Fragen zum Code diskutieren können.

Ich denke du beziehst dich auf die Zeilen 252-260 aus [build_fflip_fw.sh](https://github.com/FreifunkLippe/site-ffl/blob/master/build_fflip_fw.sh):
```
    # If gluon directory does not exist do a fresh clone frome the Freifunk-Gluon Repo
    cd $DIR
    git clone https://github.com/freifunk-gluon/gluon.git gluon -b $RELEASE
    mkdir -p $DIR/gluon/site
    cp $DIR/sites-d1/ff/site.* $DIR/gluon/site/
    cd $DIR/gluon
    make update
fi
```
Kannst du mal zum Test die obigen Zeilen einzeln in einer Shell ausführen und nach jedem Schritt prüfen, ob er erfolgreich ausgeführt wurde? Folgende Dinge sollten erfolgreich abgearbeitet werden:

- [ ] gluon.git wird nach `$DIR/gluon` geklont
- [ ] Verzeichnis `$DIR/gluon/site` wird angelegt
- [ ] Alle `site.*`-Dateien werden nach `$DIR/gluon/site/` kopiert

Falls du dabei auf Fehlermeldungen stößt, poste sie bitte hier.

Bisher vermute ich, dass der Fehler in der folgenden Zeile liegt:
`cp $DIR/sites-d1/ff/site.* $DIR/gluon/site/`

Hier werden nur die Dateien aus `sites-d1/ff` kopiert. Kann das richtig sein? Oder heißen diese Verzeichnisse auf deinem System evtl. anders?

LG
Tronde

**Comment by collimas** [comment]
Nein, mit der Zeile ist alles o.k. Es muss nur irgendeine vailde Konfiguration für das make update vorliegen.

Der Knackpunkt dürfte in Zeile 261 liegen (sorry, ich habe diese gerade erst hinzugefügt, weil sie online noch nicht vorhanden war)

    git submodule update --remote gluon

Bei der Ausführung wird gemeldet, dass "gluon" unbekanntr sei. Folglich wird auch nichts upgedatet.

**Comment by Tronde** [comment]
Ok, ich kann das Verhalten nachstellen, wenn ich die Kommandos manuell ausführe.

Was mir noch nicht klar ist, an welcher Stelle du abbrichst und es in einem zweiten Durchgang erfolgreich durchlaufen lässt. Bei mir ist der Fehler auch bei erneuter Ausführung vorhanden. Ich bekomme folgende Ausgabe:
```
host$ git pull
You are not currently on a branch. Please specify which
branch you want to merge with. See git-pull(1) for details.

    git pull <remote> <branch>

host$ git submodule update --remote gluon
error: pathspec 'gluon' did not match any file(s) known to git.
Did you forget to 'git add'?
```
Diese Meldungen kommen natürlich unabhängig davon, ob das Verzeichnis gluon zuvor existierte oder nicht. Wie bekommst du es denn hin, dass es überhaupt durchläuft? Bei mir schlagen auch die Zeilen 242-251 fehl.

Abgesehen davon, dass der code in Zeile 260 und 261 nicht funktioniert, frage ich mich wofür wir ihn dort brauchen. Das Repository wurde doch gerade erst geklont und sollte daher mit aktuellen Daten vorhanden sein. Dies sollte uns jedoch nicht davon abhalten, die fehlerhaften Befehle zu fixen.

Ich habe aber noch eine andere Frage. Mit was für einem Editor hast du das Skript editiert? Die Shebang in Zeile 1 ist kaputt und die Datei enthält unsichtbare Steuerzeichen wie z.B. das `^M`. Dadurch kann ich das Script unter Linux nicht mehr ausführen.

**Comment by Tronde** [comment]
Hi,

auf meinem Server habe ich die Kodierung der Datei repariert und das Skript ausgeführt. Es wirft auch bei vorhandenem "gluon"-Verzeichnis Fehler, arbeitet dann aber weiter. Bisher kann ich folgende Erkenntnisse festhalten:
 1. `git pull` funktioniert nicht, weil ein Branch angegeben werden muss. Dies lässt sich leicht korrigieren durch `git pull origin $RELEASE`.
 2. `git submodule update --remote gluon` funktioniert nicht. Ich habe aber noch nicht ganz verstanden, was das überhaupt macht und warum dies notwendig ist. In dem Fall, dass man das Repository "gluon" frisch geklont wurde, ist die Ausführung jedoch in meinen Augen unnötig.

**Comment by collimas** [comment]
Hallo Jörg,

eine genaue Stelle kann ich gar nicht spezifizieren. Ich breche meistens
nach 1-2 Minuten Laufzeit ab, wenn ich sehen, dass das Haupt-Repositiory
fertig geklont wurde.

Das ist aber ein manueller Vorgang, der jedes Mal an leicht anderer
Stelle passieren kann.

Meistens arbeite ich mit Notepad++ oder dem Editor von WinSCP.

Ich habe einen neuen Pull-Request erstellt und dort meine aktuelle
.sh-Datei hochgeladen. Möglicherweise sind dort keine Fehler enthalten;
bei mir läuft diese.

Mit freundlichen Grüßen
Best regards

Michael Brinkmann

Threema ID: 2TW6WHXH

Am 07.09.2016 um 21:50 schrieb Joerg Kastning:

**Comment by Tronde** [comment]
Hallo Michael,

prüfe bitte, ob dein Notepad++ die Zeilenschaltung im Unix-Format verwendet und die Zeichenkodierung UTF-8. Andernfalls kann es unter Linux zu Fehlern bei der Skriptverarbeitung kommen.

**Comment by Tronde** [comment]
Habe das Skript zum Bau der Firmware bei vorhandenem gluon-Verzeichnis ausgeführt. Es wurden keine Firmware-Dateien erzeugt. Leider habe ich keine Ursache dafür finden können.

Im nächsten Versuch entferne ich das vorhandene gluon-Verzeichnis und starte das Skript erneut.
